src/components/JsonTreeView.tsx

```
@@ -283,18 +305,18 @@ const TreeNode: React.FC<TreeNodeProps> = ({ // Destructure all props
   const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
-      // Check if the related target (where focus is going) is one of the control buttons
-      // If it is, don't submit yet, let the button click handle it.
-      const relatedTarget = e.relatedTarget as HTMLElement | null;
-      if (relatedTarget && relatedTarget.closest('.edit-control-button')) {
-          return;
-      }
-      onEditSubmit(path, convertValueType(editValue)); // Submit on blur otherwise
+    // Check if the related target (where focus is going) is one of the control buttons
+    // If it is, don't submit yet, let the button click handle it.
+    const relatedTarget = e.relatedTarget as HTMLElement | null;
+    if (relatedTarget && relatedTarget.closest('.edit-control-button')) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)